src/probnum/diffeq/perturbedsolvers/perturbedstepsolver.py

```diff
@@ -0,0 +1,91 @@
+"""ODE solver as proposed by Abdulle and Garegnani."""
+import numpy as np
+
+from probnum import diffeq, randvars
+from probnum.diffeq import wrappedscipysolver
+from probnum.diffeq.perturbedsolvers import (
+    _perturbation_functions,
+    perturbedstepsolution,
+)
+from probnum.type import FloatArgType
+
+
+class PerturbedStepSolver(diffeq.ODESolver):
+    """ODE Solver based on Scipy that introduces uncertainty by perturbing the time-
+    steps."""
```

Comment:
does it have to be based on scipy? Any diffeq.ODESolver should work, right?

Reply:
aaah I see, the problem lies in the dense output, right? The perturbed step solution wants a stack of dense output types, but the probnum dense output is a single object. This is why things aren't as compatible as they should. Do I understand correctly?

Reply:
Yes, that's correct. At the moment, the other solvers can only use the perturbation functions.

Reply:
do you think it would be worth writing a "feature request' issue about that? I have a feeling that by reconsidering parts of the ODE dense output, things like the random step solver could be unlocked pretty easily.

Reply:
can this be resolved?